src/Sentry/SentryLogger.php

```
@@ -52,49 +66,49 @@ public function setAllowedPriority(array $allowedPriority): void
 	 * @param mixed $message
 	 */
 	public function log($message, string $priority = ILogger::INFO): void
 	{
-		if (!in_array($priority, $this->allowedPriority, true)) {
+		if (in_array($priority, $this->allowedPriority, true)
+			&& ($message instanceof \Throwable)) {
+			$this->sendError($message, [
+				'level' => $this->mapLevel($priority),
+			]);
 			return;
 		}
 
-		$level = $this->getLevel($priority);
-		if ($level === null) {
-			return;
-		}
-
-		$data = [
-			'level' => $level,
-		];
-
-		$this->makeRequest($message, $data);
+		// all other messages:
+		// todo: does this really make sense? Shouldn't be added everything to breadcrumbs and errors handled separately?
+		// todo: this would probably make more sense, as breacrumbs are sent nowhere when no error happens
+		// todo: and it makes send to have also things from not allowed levels, as only last 100 breadcrums are associated with an error
+		$this->getClient()->breadcrumbs->record([
+			'message' => (string) $message,
+			'category' => 'logged-message',
+			'level' => $this->mapLevel($priority),
+		]);
 	}
 
 	/**
 	 * @param mixed $message
 	 * @param mixed[] $data
 	 */
-	protected function makeRequest($message, array $data): void
+	protected function sendError($message, array $data): void
 	{
-		$client = new Raven_Client(
-			$this->configuration[self::CONFIG_URL],
-			$this->configuration[self::CONFIG_OPTIONS]
-		);
-		if ($message instanceof Throwable) {
-			$client->captureException($message, $data);
+		if ($message instanceof \Throwable) {
+			$this->getClient()->captureException($message, $data);
 		} else {
-			$client->captureMessage($message, [], $data);
+			$this->getClient()->captureMessage($message, [], $data);
 		}
 	}
 
-	protected function getLevel(string $priority): ?string
+	private function mapLevel(string $priority): string
 	{
-		return self::LEVEL_PRIORITY_MAP[$priority] ?? null;
+		return self::LEVEL_PRIORITY_MAP[$priority] ?? Raven_Client::INFO;
```

If someone provide wrong level, there are two valid reactions:

1. log it always (e.g. assigned highest priority)
2. throw an expcetion (same as 1.), but also kills app
3. original behavirour: silently do nothing (I do not like this as you do not know that something bad happend)

---

Imho this function should not act as a fail-back. The Tracy levels are probably not going to change so often and in case they do, this function should fail on undefined offset - broadcast the error and not try to hide it.